x/lockup/keeper/lock_test.go

```diff
@@ -877,6 +1001,90 @@ func (suite *KeeperTestSuite) TestSlashTokensFromLockByID() {
+	clPool := suite.PrepareConcentratedPool()
+	positionID, _, _, liquidity, _, concentratedLockId, err := suite.App.ConcentratedLiquidityKeeper.CreateFullRangePositionLocked(suite.Ctx, clPool, addr, defaultPositionCoins, time.Hour)
+
+	// Refetch the cl pool post full range position creation
+	clPool, err = suite.App.ConcentratedLiquidityKeeper.GetPoolFromPoolIdAndConvertToConcentrated(suite.Ctx, clPool.GetId())
+	suite.Require().NoError(err)
+
+	clPoolPositionDenom := cltypes.GetConcentratedLockupDenom(clPool.GetId(), positionID)
+
+	// Store the cl pool balance before the slash
+	clPoolBalancePreSlash := suite.App.BankKeeper.GetAllBalances(suite.Ctx, clPool.GetAddress())
+
+	// Check the period locks accumulation
+	acc := suite.App.LockupKeeper.GetPeriodLocksAccumulation(suite.Ctx, types.QueryCondition{
+		Denom:    clPoolPositionDenom,
+		Duration: time.Second,
+	})
+	suite.Require().Equal(liquidity.TruncateInt64(), acc.Int64())
+
+	// The lockup module account balance before the slash should match the liquidity added to the lock
+	lockupModuleBalancePreSlash := suite.App.LockupKeeper.GetModuleBalance(suite.Ctx)
+	suite.Require().Equal(sdk.NewCoins(sdk.NewCoin(clPoolPositionDenom, liquidity.TruncateInt())), lockupModuleBalancePreSlash)
+
+	// Slash 10000000 cl/pool/1/1 and the underlying assets
+	// Figure out the underlying assets from the liquidity slash
+	position, err := suite.App.ConcentratedLiquidityKeeper.GetPosition(suite.Ctx, positionID)
+	suite.Require().NoError(err)
+	previousPositionLiquidity := position.Liquidity
+	position.Liquidity = sdk.NewDec(10000000)
+	concentratedPool, err := suite.App.ConcentratedLiquidityKeeper.GetPoolFromPoolIdAndConvertToConcentrated(suite.Ctx, position.PoolId)
+	suite.Require().NoError(err)
+	asset0, asset1, err := cl.CalculateUnderlyingAssetsFromPosition(suite.Ctx, position, concentratedPool)
+	suite.Require().NoError(err)
+	underlyingAssetsToSlash := sdk.NewCoins(asset0, asset1)
+
+	// The expected new liquidity is the previous liquidity minus the slashed liquidity
+	expectedNewLiquidity := sdk.NewCoin(clPoolPositionDenom, previousPositionLiquidity.Sub(sdk.NewDec(10000000)).TruncateInt())
+
+	// Slash the tokens from the lock
+	suite.App.LockupKeeper.SlashTokensFromLockByIDSendUnderlyingAndBurn(suite.Ctx, concentratedLockId, sdk.Coins{sdk.NewInt64Coin(clPoolPositionDenom, position.Liquidity.TruncateInt64())}, underlyingAssetsToSlash, clPool.GetAddress())
+	acc = suite.App.LockupKeeper.GetPeriodLocksAccumulation(suite.Ctx, types.QueryCondition{
+		Denom:    clPoolPositionDenom,
+		Duration: time.Second,
+	})
+	suite.Require().Equal(expectedNewLiquidity.Amount.Int64(), acc.Int64())
+
+	// The lockup module account balance after the slash should match the liquidity minus the slashed liquidity
+	lockupModuleBalancePostSlash := suite.App.LockupKeeper.GetModuleBalance(suite.Ctx)
+	suite.Require().Equal(sdk.NewCoins(expectedNewLiquidity), lockupModuleBalancePostSlash)
```

Comment:
We should also be checking the fact of burning or supply decreasing

Reply:
https://github.com/osmosis-labs/osmosis/pull/4884/commits/731d107b7595c46226d18b0630f1533650c47471